bubblegum/program/src/lib.rs

```diff
@@ -935,69 +827,39 @@ pub mod bubblegum {
-        Ok(())
-    }
-
-    pub fn request_mint_authority(ctx: Context<SetMintRequest>, mint_capacity: u64) -> Result<()> {
-        let request = &mut ctx.accounts.mint_authority_request;
-        assert_enough_mints_to_approve(&ctx.accounts.tree_authority, mint_capacity)?;
-        request.init_or_set(ctx.accounts.mint_authority.key(), mint_capacity);
-        Ok(())
-    }
-
-    pub fn approve_mint_authority_request(
-        ctx: Context<ApproveMintRequest>,
-        num_mints_to_approve: u64,
-    ) -> Result<()> {
-        let authority = &mut ctx.accounts.tree_authority;
-        let request = &mut ctx.accounts.mint_authority_request;
-        // Check that there are enough valid mints left in tree to approve
-        assert_enough_mints_to_approve(authority, num_mints_to_approve)?;
-        authority.approve_mint_capacity(num_mints_to_approve);
-        request.approve(num_mints_to_approve)?;
-        Ok(())
-    }
-
-    pub fn close_mint_request(ctx: Context<CloseMintRequest>) -> Result<()> {
-        let authority = &mut ctx.accounts.tree_authority;
-        let request = &ctx.accounts.mint_authority_request;
-        // Transfer remaining mint capacity to authority
-        authority.restore_mint_capacity(request.num_mints_approved);
-        Ok(())
-    }
-
     pub fn set_tree_delegate(ctx: Context<SetTreeDelegate>) -> Result<()> {
         ctx.accounts.tree_authority.delegate = ctx.accounts.new_delegate.key();
         Ok(())
     }
 
     pub fn mint_v1(ctx: Context<MintV1>, message: MetadataArgs) -> Result<()> {
         // TODO -> Separate V1 / V1 into seperate instructions
+        let payer = ctx.accounts.payer.key();
+        let incoming_tree_delegate = ctx.accounts.tree_delegate.key();
+        let tree_creator = ctx.accounts.authority.creator;
+        let tree_delegate = ctx.accounts.authority.delegate;
         let owner = ctx.accounts.owner.key();
         let delegate = ctx.accounts.delegate.key();
-        let mint_authority = &mut ctx.accounts.mint_authority;
+        let authority = &mut ctx.accounts.authority;
         let merkle_tree = &ctx.accounts.merkle_tree;
 
-        // The mint authority must sign if it is not equal to the tree authority.  Also, if the
-        // mint authority is a signer it can be used for creator validation.
-        let mut metadata_auth = HashSet::<Pubkey>::new();
-        if mint_authority.key() != ctx.accounts.authority.key() {
-            require!(
-                mint_authority.is_signer,
-                BubblegumError::MintAuthorityMustSign
-            );
-            metadata_auth.insert(mint_authority.key());
+        require!(
+            incoming_tree_delegate == tree_creator || incoming_tree_delegate == tree_delegate,
+            BubblegumError::TreeAuthorityIncorrect,
+        );
+
+        if !authority.contains_mint_capacity(1) {
+            return Err(BubblegumError::InsufficientMintCapacity.into());
         }
```

Comment:
```suggestion
       require(authority.contains_mint_capacity(1), BubblegumError::InsufficientMintCapacity);
```

NB: Is this not equivalent?

Reply:
To me it seems about equivalent, I'll probably move towards require statement in later PR.  Thanks!